criu/files-reg.c

```diff
@@ -1149,18 +1152,39 @@ static int create_link_remap(char *path, int len, int lfd, u32 *idp, struct ns_i
 	rfe.fown = &fwn;
 	rfe.name = link_name + 1;
 
-	/* Any 'unique' name works here actually. Remap works by reg-file ids. */
-	snprintf(tmp + 1, sizeof(link_name) - (size_t)(tmp - link_name) - 1, "link_remap.%d", rfe.id);
-
 	mntns_root = mntns_get_root_fd(nsid);
 
+	fd = open("/dev/urandom", O_RDONLY);
```

Comment:
should we generate this uid once for the entire dump?

Reply:
>This change introduces generating a random suffix to prevent link_remap collisions when a dump fails and another dump is attempted on the same process

> should we generate this uid once for the entire dump?

We already do this for [`dump_criu_run_id`](https://github.com/checkpoint-restore/criu/blob/b24f6e25bb99211ddfe971bb7f47778928a2f737/criu/util.c#L2036). Can we use this value instead of generating a new one?